Code2Gether-Discord-Bot.Library/BusinessLogic/ExcuseGeneratorLogic.cs

```diff
@@ -0,0 +1,88 @@
+﻿using Code2Gether_Discord_Bot.Library.Models;
+using Discord;
+using Discord.Commands;
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Code2Gether_Discord_Bot.Library.BusinessLogic
+{
+    public class ExcuseGeneratorLogic : IBusinessLogic
+    {
+        #region Fields
+        private ILogger _logger;
+        private ICommandContext _context;
+        #endregion
+
+        #region Constructor
+        public ExcuseGeneratorLogic(ILogger logger, ICommandContext context)
+        {
+            _logger = logger;
+            _context = context;
+        }
+        #endregion
+
+        public Embed Execute()
+        {
+            _logger.Log(_context);
+
+            Embed embed;
+
+            try
+            {
+                if (!_context.Guild.GetCurrentUserAsync().Result.GuildPermissions.ManageMessages)
```

Comment:
In this command, requiring the ManageMessages role for this to work isn't a good design choice. In the Ping command it was needed to calculate timestamps. You can simply omit the `temporaryMessage` completely